src/hotspot/share/gc/shared/genCollectedHeap.cpp

```diff
@@ -791,16 +791,19 @@ void GenCollectedHeap::process_roots(ScanningOption so,
 
 void GenCollectedHeap::full_process_roots(bool is_adjust_phase,
                                           ScanningOption so,
-                                          bool only_strong_roots,
                                           OopClosure* root_closure,
                                           CLDClosure* cld_closure) {
   // Called from either the marking phase or the adjust phase.
   const bool is_marking_phase = !is_adjust_phase;
 
-  MarkingCodeBlobClosure mark_code_closure(root_closure, is_adjust_phase, is_marking_phase);
-  CLDClosure* weak_cld_closure = only_strong_roots ? NULL : cld_closure;
-
-  process_roots(so, root_closure, cld_closure, weak_cld_closure, &mark_code_closure);
+  if (is_marking_phase) {
```

Comment:
I do not think it is useful to explicitly create a local variable of the inverse of a parameter for its single use. That looks like the parameter is wrong.

I recommend either using `is_adjust_phase` directly here (potentially swapping the branches) or change the parameter of the method and the callers if you prefer to have `is_marking_phase` here.

Reply:
After this change, it becomes apparent this method should be split into two, for marking/adjust phase respectively, which I plan to address in the next PR.

Reply:
I would not mind doing this in this PR - it is called `Refactor GenCollectedHeap::full_process_roots` after all, but I do not mind.